lib/puppet/pops/validation/checker4_0.rb

```diff
@@ -530,6 +532,102 @@ def internal_check_future_reserved_word(o, name)
     end
   end
 
+  def internal_check_file_namespace(o, name, file)
+    return if file.nil? || file == '' #e.g. puppet apply -e '...'
+
+    file_namespace = namespace_for_file(file)
+    return if file_namespace == :no_namespace
```

Comment:
Can you make constants for the `:no_namespace` and `:bad_module_file` symbols? Otherwise it's too easy to make a typo, e.g. `:no_namspace`